app/controllers/api_controller.rb

```diff
@@ -1,96 +1,17 @@
-            cad: product.your_price_net(account, :cad, :unit)
-          },
-          piece: {
-            usd: product.your_price_net(account, :usd, :piece),
-            cad: product.your_price_net(account, :cad, :piece)
-          },
-          halfpiece: {
-            usd: product.your_price_net(account, :usd, :halfpiece),
-            cad: product.your_price_net(account, :cad, :halfpiece)
-          },
-          customer: {
-            unit: product.order_price__net(current_user, :unit),
-            piece: product.order_price__net(current_user, :piece),
-            halfpiece: product.order_price__net(current_user, :halfpiece)
-          }
-        }
-      },
-      stock: product.stock(current_user)&.merge(unit: product.measured_unit)
-    }
-
-    output[:uom_display_text] = product.measured_unit["uom_display_text"] if product.measured_unit["uom_display_text"].present?
-
-    if product.wallcovering?
-      output[:measured_unit] = product.measured_unit["long"]["singular"].downcase
-      output[:order_increment] = product.wallcovering_data.average_bolt.to_f
-    end
-
-    output[:pricing].delete(:net) unless current_user.has_role?(:view_net_pricing_usd) || current_user.has_role?(:view_net_pricing_cad)
-    output.delete(:stock) unless current_user.has_role?(:view_stock)
-
-    @product = product
-    @stock_output = output[:stock].deep_symbolize_keys
-
-    @stock_total = @stock_output[:current][:total]
-    @stock_unit = @stock_output[:unit][:long][@stock_total == 1 ? :singular : :plural]
-
-    @stock_total_reserved = @stock_output[:total_reserved]
-    @stock_total_reserved_unit = @stock_output[:unit][:long][@stock_total_reserved == 1 ? :singular : :plural]
-
-    @has_stock = @stock_total.present? && @stock_total.to_f.positive?
-    @has_availability = @stock_output[:availability].present?
-    @incoming_stock = @stock_output[:expected]
-
-    # Get all bolts and separate them into "regular bolts" and "small cuts"
-    @bolts = @stock_output[:current][:bolts]
-
-    # Small cuts applies to fabric sold by the yard, excluding Clarencehouse
-    if !clarencehouse? && @product.fabric? && @product.measured_unit["long"]["singular"] == "Yard"
-      @regular_bolts, @small_cuts = @bolts.partition { |_bolt, lot| lot[:quantity].to_f >= 5 } if @bolts.present?
-    else
-      @regular_bolts = @bolts
-      @small_cuts = nil
-    end
-
-    # Group and sort by dye lot number
-    @regular_bolts_dye_lots = @regular_bolts.group_by { |_bolt, lot| lot[:dye_lot] } if @regular_bolts.present?
-    @small_cuts_dye_lots = @small_cuts.group_by { |_bolt, lot| lot[:dye_lot] } if @small_cuts.present?
+    ApiHelper.calculate(Product.find(params[:sku]), current_account, current_user, clarencehouse?)
+    build_view_attributes
   rescue => e
     render json: { success: false, error: e.message }, status: :unprocessable_entity
   end
+
+  private
+
+  def build_view_attributes
+    ApiHelper.class_variables.map { |cv| eval("#{cv.to_s[1..-1]} = #{cv}") }
```

Comment:
If are looking at this line and haven't read my notes, please give these a read to explain how I landed here. 
[work_in_progress.pdf](https://github.com/fabricut/interview-app/files/10018020/work_in_progress.pdf)